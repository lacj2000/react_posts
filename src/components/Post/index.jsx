const Post = ({ url, title, body }) => (
    <div className="post">
        <img src={url} alt={title} />
        <div className="post-content">
            <h2>{title}</h2>
            <p>{body}</p>
        </div>
    </div>
);


export default Post;