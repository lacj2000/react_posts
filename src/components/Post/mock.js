export const postPropsMock = {
    url: 'img/test.png',
    title: 'test title',
    body: 'test body',
}