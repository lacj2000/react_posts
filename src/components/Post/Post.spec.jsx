import { render, screen } from "@testing-library/react"
import Post from "."
import { postPropsMock } from "./mock"

const props = postPropsMock

describe('<Post />', () => {
    it('should render Post correctly',()=>{
        render(<Post {...props} />);
        expect(screen.getByRole('img',{ src: props.url } ))
          .toHaveProperty('alt', props.title);
        expect(screen.getByRole('heading',{ name: props.title } ))
          .toBeInTheDocument();
        expect(screen.getByText(props.body)).toBeInTheDocument();
        
    });
    it('should match snapshot',()=>{
        const { container } = render(<Post {...props} />);
        expect(container.firstChild).toMatchSnapshot(); 
    });


});