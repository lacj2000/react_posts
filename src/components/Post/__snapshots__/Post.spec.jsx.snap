// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="test title"
    src="img/test.png"
  />
  <div
    class="post-content"
  >
    <h2>
      test title
    </h2>
    <p>
      test body
    </p>
  </div>
</div>
`;
